### PR TITLE
Fourth layer

### PR DESCRIPTION
**Pull request by DariaKisielewska**
- original PR taken from Sushil (modifiations from Krzysztof)
- implementing modular layer (24 moduls; each 13 scintillators)
- (fixed) each scinitllator have unique id

### CODE REVIEW

File: src/DetectorConstruction.cc [hunk 1]
```diff
@@ -384,13 +429,23 @@ void DetectorConstruction::ConstructFrameCAD()
 
 }
 
+G4int DetectorConstruction::ReturnNumberOfScintillators()
+{
+      if(fLoadModularLayer){
+            return 504;
+      } else {
+            return 192;
```
Comment (kkacprzak) [inline src/DetectorConstruction.cc]: Is this has to be hardcoded?

File: include/DetectorConstruction.hh [hunk 2]
```diff
@@ -34,24 +34,33 @@ class DetectorConstructionMessenger;
...
 
 
+// parameters for modular layer (known as 4th layer)
+const G4double scinDim_x_inModule = 2.5*cm; ///<  X dimension of simulated strip
+const G4double scinDim_y_inModule = 0.6*cm; ///<  Y dimension of simulated strip
 
-const G4double radius[layers] = {42.5*cm,46.75*cm,57.5*cm}; ///< layer radius up to center of the strip
-const int nSegments[layers] = {48,48,96}; ///< number of segments in each layer
-//const int nSegments[layers] = {2,2,2}; ///< number of segments in each layer
+const int modulesInModularLayer = 24;
```
Comment (kkacprzak) [inline include/DetectorConstruction.hh]: Maybe all the constant parameters can be put in some configuration list or class?

**Review by kkacprzak — APPROVED** [review]
I have a general development suggestion: can all scintillator objects be made in to separate class, that is then used in DetectorConstruction? Then stored in some array, so it is possible to get number of them, and so on? Also lot of constants are all over the project, maybe it would be easier to keep them in one place/separate class?

**Comment by Dr-Sushil-Sharma** [comment]
I also agree with the opinion of Krzysztof. We have to somehow save the constants separately. I also would like to ask about the possibility to assign the reconfiguration of 4th layer in different combinations. Are we considering it?? One layer with 24 modules or 3 layer (4,8,12) with 24 modules.... I think it will add up  few more constants. We may discuss this in next meeting dedicated to Monte Carlo.

**Comment by DariaKisielewska** [comment]
Additional configurations will  be pushed in separate PR (if needed). 
Sushil, please review the PR

**Review by Dr-Sushil-Sharma — APPROVED** [review]

**Comment by wkrzemien** [comment]
retest me please

**Pull request merged**